js/src/bitswap.js

```diff
@@ -0,0 +1,137 @@
+    let ipfsA
+    let ipfsB
+    let ipfsBId
+    const key = 'QmUBdnXXPyoDFXj3Hj39dNJ5VkN3QFRskXxcGaYFBB8CNR'
+
+    before(function (done) {
+      // CI takes longer to instantiate the daemon, so we need to increase the
+      // timeout for the before step
+      this.timeout(60 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+        series([
+          (cb) => factory.spawnNode((err, node) => {
+            expect(err).to.not.exist()
+            ipfsA = node
+            cb()
+          }),
+          (cb) => factory.spawnNode((err, node) => {
+            expect(err).to.not.exist()
+            ipfsB = node
+            cb()
+          }),
+          (cb) => {
+            ipfsB.id((err, id) => {
+              expect(err).to.not.exist()
+              const ipfsBAddr = id.addresses[0]
+              ipfsBId = id.id
+              ipfsA.swarm.connect(ipfsBAddr, cb)
+            })
+          },
+          (cb) => {
+            ipfsB.block.get(new CID(key))
+              .then(() => {})
+              .catch(() => {})
+            setTimeout(cb, 500)
+          }
+        ], done)
+      })
+    })
+
+    after((done) => common.teardown(done))
+
+    it('.stat', (done) => {
+
+      ipfsA.bitswap.stat((err, stats) => {
+        statsTests.expectIsBitswap(err, stats)
+        done()
+      })
+    })
+
+    it('.wantlist', (done) => {
+      ipfsB.bitswap.wantlist((err, list) => {
+        expect(err).to.not.exist()
+        expect(list[0].cid.toBaseEncodedString()).to.equal(key)
+        done()
+      })
+    })
+
+    it('.wantlist peerid', (done) => {
+      ipfsA.bitswap.wantlist(ipfsBId, (err, list) => {
+        expect(err).to.not.exist()
+        expect(list[0].cid.toBaseEncodedString()).to.equal(key)
+        done()
+      })
+    })
+
+    it('.unwant', (done) => {
+      ipfsA.bitswap.unwant(new CID(key), (err) => {
+        ipfsA.bitswap.wantlist((err, list) => {
+          expect(list).to.be.empty()
+          done()
+        })
+      })
+    })
+  })
+
+  describe('.bitswap offline', () => {
+    let ipfs
+
+    before(function (done) {
+      // CI takes longer to instantiate the daemon, so we need to increase the
+      // timeout for the before step
+      this.timeout(60 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist()
+          ipfs = node
+          ipfs.id((err, id) => {
+            expect(err).to.not.exist()
+            ipfs.stop((err) => {
+              // TODO: go-ipfs returns an error, https://github.com/ipfs/go-ipfs/issues/4078
+              if (!id.agentVersion.startsWith('go-ipfs')) {
+                expect(err).to.not.exist()
+              }
+              done()
+            })
+          })
+        })
+      })
+    })
+
+    it('.stat gives error while offline', () => {
+      ipfs.bitswap.stat((err, stats) => {
+        expect(err).to.exist()
```

Comment:
Can we assert against the error message? Just expecting it to exist could give to false positives.